en/guide/missions.md

```diff
@@ -0,0 +1,201 @@
+
+> **Note** The API enables a small but useful subset of the mission commands supported by PX4 (and the MAVLink specification). For example, it does not currently support "repeat", takeoff, return to land etc.
+
+Missions are *managed* though the [Mission](../api_reference/classdronecore_1_1_mission.md) class, which communicates with the device to upload mission information and run, pause, track the mission progress etc. The mission that is uploaded to the vehicle is defined as a vector of [MissionItem](../api_reference/classdronecore_1_1_mission_item.md) objects. 
+
+
+## Preconditions
+
+The following code assumes that you already have included DroneCore (`#include <dronecore/dronecore.h>`) and the standard library (`#include <functional>`) and that there is a [connection to a device](../guide/connections.md) obtained as shown below:
+```cpp
+Device &device = dc.device(); 
+```
+
+## Defining a Mission
+
+A mission must be defined as a vector of [MissionItem](../api_reference/classdronecore_1_1_mission_item.md) objects as shown below:
+```cpp
+std::vector<std::shared_ptr<MissionItem>> mission_items;
+```
+
+You can create as many `MissionItem` objects as you like and use `std_vector::push_back()` to add them to the back of the mission item vector. The example below shows how to create and add a `MissionItem`, that just sets the target position (using [set_position()](../api_reference/classdronecore_1_1_mission_item.md#classdronecore_1_1_mission_item_1ab5897670c8830fc3514036d6ee99b582)). 
+```cpp
+// Create MissionItem and set its position
+std::shared_ptr<MissionItem> new_item(new MissionItem());
+new_item->set_position(47.40 /*latitude_deg*/, 8.5455360114574432 /*longitude_deg*/);
+
+// Add new_item to the vector
+mission_items.push_back(new_item);
+```
+
+The example below shows how you might set the other options on a second `MissionItem` (and add it to the mission).
+
+```cpp
+std::shared_ptr<MissionItem> new_item2(new MissionItem());
+new_item2->set_position(47.50 /*latitude_deg*/, 8.5455360114574432 /*longitude_deg*/);
+new_item2->set_relative_altitude(2.0f);
+new_item2->set_speed(5.0f /* m/s */);
+new_item2->set_fly_through(true);
+new_item2->set_gimbal_pitch_and_yaw(20.0f /*pitch*/, 60.0f /*yaw degrees*/);
+new_item2->set_camera_action(MissionItem::CameraAction::TAKE_PHOTO);
+    
+//Add new_item2 to the vector
+mission_items.push_back(new_item2);
+```
+
+> **Note** The autopilot has sensible default values for the attributes. If you do set a value (e.g. the desired speed) 
+then it will be the default for the remainder of the mission.
+
+<span></span>
+> **Note** There are also getter methods for querying the current value of `MissionItem` attributes.

```

Comment:
And Note that the default values of most fields are `NaN` which means they are ignored.